### ISSUE TITLE
Sun showing Moon icon?

### ISSUE THREAD

**Issue by Mariusthvdb** [home-assistant/frontend#7672]
## Checklist

- [ ] I have updated to the latest available Home Assistant version.
- [ ] I have cleared the cache of my browser.
- [ ] I have tried a different browser to see if it is related to my browser.

## The problem

Suddenly notice Sun showing a Moon icon (on a badge) while the sun more-info still shows the weather-sunny:

<img width="129" alt="Schermafbeelding 2020-11-12 om 22 56 09" src="https://user-images.githubusercontent.com/33354141/99001147-a6b2d200-253a-11eb-877c-4d31aeef43d9.png">

<img width="383" alt="Schermafbeelding 2020-11-12 om 22 56 15" src="https://user-images.githubusercontent.com/33354141/99001173-af0b0d00-253a-11eb-91c2-9b9bcdd4d17d.png">

## Expected behavior

simply use the original weather-sunny and weather-night, or, if state change could set the icon maybe use weather-sunset-down/up

## Environment

- Home Assistant release with the issue: 117/118b
- Last working Home Assistant release (if known): ?
- Browser and browser version: Chrome, Safari
- Operating system: Mac OS

## State of relevant entities

```yaml
badges:
  - binary_sensor.updater
  - sensor.ha_local_version
  - sensor.moon
  - sun.sun
```
<img width="818" alt="Schermafbeelding 2020-11-12 om 23 02 13" src="https://user-images.githubusercontent.com/33354141/99001429-2476dd80-253b-11eb-9e38-f35bc86c34ff.png">

## Problem-relevant configuration

```yaml

```
sun:

## Javascript errors shown in your browser console/inspector

```txt

```
none
## Additional information

using the sun in a glance card:
```
  - type: custom:hui-element
    card_type: glance
    style: |
      ha-card {
        box-shadow: none;
        margin: 0px -16px 0px -16px;
      }
    entities:
      - sun.sun
      - sensor.moon
      - entity: sensor.season_astronomical
        name: Season

```

shows fine:

<img width="435" alt="Schermafbeelding 2020-11-12 om 23 06 44" src="https://user-images.githubusercontent.com/33354141/99001806-c72f5c00-253b-11eb-9f2b-f7f130b24e26.png">

showing weather-night correctly

**Comment by spacegaier** [comment]
Are you suing a custom theme that is overriding some icons? I am asking because, I can reproduce the visuals of the badge and the more-info, but in a glance card I (consistent with the more-info) also get the sun icon, and not a moon with stars.

**Comment by Mariusthvdb** [comment]
no, no custom theme overriding icons, but I forgot to post the customization I use on the main system, 
```
    sun.sun:
      templates:
        icon: >
          if (state =='above_horizon') return 'mdi:weather-sunny';
          return 'mdi:weather-night';
```



so you're right, that shouldn't be in this issue.... 

main issue remains though as you've established, and these are non customized, as core as can be.

**Comment by Mariusthvdb** [comment]
still happening on release 118.

please reopen?

<img width="138" alt="Schermafbeelding 2020-11-19 om 21 00 26" src="https://user-images.githubusercontent.com/33354141/99718005-7a530480-2aaa-11eb-8775-a94c52490157.png">
<img width="390" alt="Schermafbeelding 2020-11-19 om 21 00 36" src="https://user-images.githubusercontent.com/33354141/99718008-7aeb9b00-2aaa-11eb-830d-b9df6415a508.png">

**Comment by spacegaier** [comment]
My fix was not picked for 0.118. So it is in the dev branch only.

**Comment by Mariusthvdb** [comment]
a ok, thanks, will await the next release update then.
